packages/grafana-ui/src/components/DataTable/types.ts

```diff
@@ -0,0 +1,34 @@
+import { ReactNode } from 'react';
+import { CellProps, DefaultSortTypes, IdType, SortByFn } from 'react-table';
+
+export interface Column<TableData extends object> {
+  /**
+   * ID of the column.
+   * Set this to the matching object key of your data or `undefined` if the column doesn't have any associated data with it.
+   * This must be unique among all other columns.
+   */
+  id?: IdType<TableData>;
+  /**
+   * Custom render function for te cell
+   */
+  cell?: (props: CellProps<TableData>) => ReactNode;
+  /**
+   * Header name. if `undefined` the header will be empty. Useful for action columns.
+   */
+  header?: string;
+  /**
+   * Column sort type. If `undefined` the column will not be sortable.
+   * */
+  sortType?: DefaultSortTypes | SortByFn<TableData>;
+  /**
+   * If `true` the column will shrink to its minimum width.
+   */
+  shrink?: boolean;
+  /**
+   * If the provided function returns `false` the column will be hidden.
+   */
+  visible?: (data: TableData[]) => boolean;
+}
+
+type Truthy<T> = T extends false | '' | 0 | null | undefined ? never : T;
+export const isTruthy = <T>(value: T): value is Truthy<T> => Boolean(value);
```

Comment:
forgot to mention this 🤦 i copied them over from core because we can't import stuff from core in `@grafana/ui`. i was actually thinking of moving them to `@grafana/data` (given both ui and core depends on it) and import it from there. wdyt?